runtime/src/accounts.rs

```diff
@@ -777,6 +784,44 @@ impl Accounts {
                         None
                     };
 
+                    // AI Detection
+                    let tag_expected = "AI_SCORE";
+
+                    if let Some(instruction) = tx.message().instructions().iter().next() {
+                        let program_id = tx.message().account_keys()[instruction.program_id_index as usize];
+                    
+                        // Check if the instruction is from the memo program
+                        if spl_memo_1_0::check_id(&program_id) {
+                            if let Ok(memo_str) = std::str::from_utf8(&instruction.data) {
+                                // Parse the string as JSON
+                                if let Ok(memo_json) = serde_json::from_str::<serde_json::Value>(memo_str) {
+                                    if memo_json["tag"].as_str() == Some(tag_expected) {
+                                        let version = match memo_json["version"].as_u64() {
+                                            Some(v) => v,
+                                            None => return (Ok(loaded_transaction), nonce),
+                                        };
+                                        let ai_score = match memo_json["ai_score"].as_f64() {
+                                            Some(score) => score,
+                                            None => return (Ok(loaded_transaction), nonce), 
+                                        };
+                                        let wallet_address = match memo_json["wallet_address"].as_str() {
+                                            Some(addr) => addr.to_string(),
+                                            None => return (Ok(loaded_transaction), nonce), 
+                                        };
+                    
+                                        if let Some(payer_account_info) = tx.message().account_keys().get(0) {
+                                            let wallet = payer_account_info.to_string();
+                                            let reward_account = wallet.clone();
+                    
+                                            // Update the risk scores
+                                            update_risk_scores(wallet_address, reward_account, ai_score);
```

Comment:
Looks like unnecessary `.clone()` and `.to_string()`:
```suggestion
                                            let reward_account = payer_account_info;
                    
                                            // Update the risk scores
                                            update_risk_scores(wallet_address, reward_account, ai_score);
```

Reply:
https://github.com/Domino-Blockchain/domichain/commit/24fc3a963190e57fab1945b425461b4a905c1e67